Core/src/main/java/org/gusdb/fgputil/DualBufferBinaryRecordReader.java

```diff
@@ -92,9 +85,7 @@ public Optional<byte[]> next() {
       }
     }
     // read the next record
-    byte[] nextRecord = _recordBufferSupplier.get();
```

Comment:
I think my hesitation with this approach before is that the function has no enforced obligation to take only the number of bytes passed to the constructor as a record size.  This breaks the encapsulation of this class.  Maybe is ok, but should be well documented in the constructor.  I wonder if we can do a quick check to ask what index the ByteBuffer is at before an after it is sent to the Function, then throw IllegalStateException if the function consumes more bytes than it should?